src/EnergyPlus/UnitarySystem.cc

```diff
@@ -6657,22 +6706,53 @@ namespace UnitarySystems {
                     this->m_NumOfSpeedHeating = this->m_CompPointerMSHP->numOfSpeedHeating;
                     this->m_HeatMassFlowRate.resize(this->m_NumOfSpeedHeating + 1);
                     this->m_HeatVolumeFlowRate.resize(this->m_NumOfSpeedHeating + 1);
                     this->m_MSHeatingSpeedRatio.resize(this->m_NumOfSpeedHeating + 1);
                     if (state.dataGlobal->DoCoilDirectSolutions && this->m_NumOfSpeedCooling < this->m_NumOfSpeedHeating) {
                         this->FullOutput.resize(this->m_NumOfSpeedHeating + 1);
                     }
+                    if (this->m_HeatingCoilType_Num == DataHVACGlobals::Coil_HeatingWaterToAirHPSimple && this->m_NumOfSpeedHeating > 1) {
+                        this->m_MultiSpeedHeatingCoil = true;
+                        this->m_MultiOrVarSpeedHeatCoil = true;
+                    }
                     for (int i = 1; i <= this->m_NumOfSpeedHeating; ++i) {
                         this->m_HeatMassFlowRate[i] = 0.0;
                         this->m_HeatVolumeFlowRate[i] = 0.0;
                         this->m_MSHeatingSpeedRatio[i] = 1.0;
                     }
+                    if (this->m_HeatingCoilType_Num == DataHVACGlobals::Coil_HeatingWaterToAirHPVSEquationFit) {
+                        std::string MultispeedType = "UnitarySystemPerformance:Multispeed";
+                        if (this->m_DesignSpecMSHPIndex == -1) {
+                            std::string MultispeedType = "Fan:SystemModel";
+                        }
+                        int NumOfSpeed = VariableSpeedCoils::GetVSCoilNumOfSpeeds(state, this->m_HeatingCoilName, errorsFound);
+                        if (errorsFound) {
+                            ShowSevereError(state,
+                                            format("{} = {} is not found. Please check the input.", cCurrentModuleObject, this->m_HeatingCoilName));
+                        }
+                        if (NumOfSpeed != this->m_NumOfSpeedHeating) {
+                            ShowWarningError(state, format("{} = {}.", cCurrentModuleObject, this->m_HeatingCoilName));
+                            ShowContinueError(state,
+                                              format("... The number of heating coil speeds in the {} = {:.0R}",
+                                                     MultispeedType,
+                                                     double(this->m_NumOfSpeedHeating)));
+                            ShowContinueError(
+                                state,
+                                format("... The number of heating coil speeds in Coil:Heating:WaterToAirHeatPump:VariableSpeedEquationFit = {:.0R}",
+                                       double(NumOfSpeed)));
+                            ShowContinueError(state, format("... The number of heating coil speeds in the {} will be used.", MultispeedType));
+                        }
+                    }
```

Comment:
Error check for variable speed HP heating coil with equation fit

Reply:
I wonder how nice this would look as a switch/case block instead of nested IFs.